docs/release-cycle.md

```diff
@@ -0,0 +1,81 @@
+
+Kubernetes has a well defined release process that improves with every release.
+These notes are collected during the v1.8 release cycle, and provide some inspiration
+for how the Kubernetes developing flow actually can look like.
+
+Most of these tips and tricks are general and not kubeadm specific.
+But as we move more and more components out of the main Kubernetes repo, these communication
+channels and escalation paths become increasingly important. Some of them are explained
+in this doc.
+
+Some of what's outlined below are just rough guidelines and processes that we use when developing kubeadm.
+It makes sense to document this so the rest of the community can see what it looks like to be involved
+in developing a core component, and/or to onboard new contributors to the process.
+
+That said, nothing here is set in stone. See this as inspiration :)
+
+#### First month of the quarter (coding)
+
+ - Manage cherrypicks for the current stable branch
+   - Make sure high-priority bug reports are triaged and fixed.
+   - Create bug fixes, ensure they're merged, and cherry pick them to the release branch.
+   - Coordinate with the stable branch manager.
+ - Plan/prioritize features, testing work and bugs
+   - This is usually done in the SIG Cluster Lifecycle meetings, and a dedicated v1.X planning doc is made and referenced from the [meeting notes](https://docs.google.com/document/d/1deJYPIF4LmhGjDVaqrswErIrV7mtwJgovtLnPCDxP7U/edit#).
+   - There people sign up for working on the high level features and objectives for the release
+   - Create the right feature issues in `kubernetes/features` for what you want to accomplish for the quarter. Deadline is often the last day of the first month.
+ - Go through _every_ open issue in the kubeadm repo.
+   - As there are a small number of core maintainers working on kubeadm, there often isn't time to answer all feature requests or bugfixes, or to close old, outdated ones.
+   - As a general practice, every issue must be triaged **at least** once a quarter.
+ - Write proposals for new and larger features
+   - Often starting as a Google Doc that is easy to comment on/let others collaborate on.
+   - Get consensus on the design in the SIG
+   - Convert to markdown, send PR to `kubernetes/community` and loop in the rest of the community (other relevant SIGs, if not looped in already)
+   - Update as needed by review comments, get LGTM and merge.
+ - Start coding new features, fix bugs and increase unit testing coverage levels
+ - Ensure that the [`kubeadm-gce`](https://k8s-testgrid.appspot.com/sig-cluster-lifecycle#kubeadm-gce) is master blocking.
+   - "Master blocking" means that this is the dashboard the release team looks at when they decide whether to cut an alpha.X release or not.
+ - Cleanup work. As kubeadm supports deploying clusters one minor version below the CLI version, support for v1.7-specific workarounds may be dropped when the v1.9 cycle starts
```

Comment:
Not sure how, but I think that we have to establish a kind of standard for marking in the code where code cleanups are foreseen/necessary at every release, minimizing somehow the risk of forgetting something.. any idea?